doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -3,68 +3,234 @@
 Syntax extensions and notation scopes
 =====================================
 
-In this chapter, we introduce advanced commands to modify the way Coq
-parses and prints objects, i.e. the translations between the concrete
-and internal representations of terms and commands.
-
-The main commands to provide custom symbolic notations for terms are
-:cmd:`Notation` and :cmd:`Infix`; they will be described in the
-:ref:`next section <Notations>`. There is also a
-variant of :cmd:`Notation` which does not modify the parser; this provides a
-form of :ref:`abbreviation <Abbreviations>`. It is
-sometimes expected that the same symbolic notation has different meanings in
-different contexts; to achieve this form of overloading, Coq offers a notion
-of :ref:`notation scopes <Scopes>`.
-The main command to provide custom notations for tactics is :cmd:`Tactic Notation`.
+While, it's in theory possible to write down every Gallina term
+explicitly, in the way they are understood by the kernel, for instance
+:n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`, this would be very
+tedious and error-prone and take us away from our usual mathematical
```

Comment:
Unlike academic papers, a chapter introduction in user documentation should first answer the questions "Is this material worth reading now?" and "Why do I care?"  The introduction should be brief and only identify the most important capabilities/concepts (somewhat akin to an abstract but not identical).

Introductions should be written to minimize what you need to know to understand them.  We want it to be accessible to people who are Coq beginners and who aren't already very familiar with Coq.  We shouldn't assume that users have an academic background in proof theory.  IMO the material should be accessible to someone with a bachelor's degree in math or computer science.

These lines introduce terms that are not essential to using notations: "Gallina", "kernel".  It would be better not to use them here.  The text is also wordy, e.g. "it's in theory possible to write down every Gallina term
explicitly, in the way they are understood by the kernel" is a lotta words.  It could be a lot more succinct, making it easier to read.

Chapter and section introductions are often critical for readers/learners to navigate the documentation.  Writing good introductions takes time and care.  And the topic sentence needs to be simple and well-crafted.